### ISSUE TITLE
[JS] schema validation error in streaming mode cause reflection API to crash

### ISSUE THREAD

**Issue by pavelgj** [firebase/genkit#271]
**Describe the bug**
When running the flow (see below) from the Dev UI in streaming mode the `genkit start` process "crashes".
Full error: https://gist.github.com/pavelgj/0ef67bc35d2c4436c7c128462a64b0b9

**To Reproduce**

```typescript
const jokeSubjectGenerator = defineTool(
  {
    name: 'jokeSubjectGenerator',
    description: 'can be called to generate a subject for a joke',
  },
  async () => {
    return 'banana';
  }
);

export const toolCaller = defineFlow(
  {
    name: 'toolCaller',
    outputSchema: z.string(),
  },
  async (_, streamingCallback) => {
    if (!streamingCallback) {
      throw new Error('this flow only works in streaming mode');
    }

    const { response, stream } = await generateStream({
      model: gemini15ProPreview,
      config: {
        temperature: 1,
      },
      tools: [jokeSubjectGenerator],
      prompt: `tell me a joke`,
    });

    for await (const chunk of stream()) {
      streamingCallback(chunk);
    }

    return (await response()).text();
  }
);
```

**Expected behavior**

**Screenshots**
![image](https://github.com/firebase/genkit/assets/774654/2a97604a-c09f-4920-8e02-b23742506ea5)

**Comment by chrisraygill** [comment]
If this was fixed in Go, do we already know the solution for JS? @pavelgj

**Comment by pavelgj** [comment]
This is from "May 29"... this specific issue is likely outdated and can be closed. 
We fixed some crash issues in 0.9.
